sentry_sdk/envelope.py

```diff
@@ -230,23 +230,25 @@ def __repr__(self):
     @property
     def data_category(self):
         # type: (...) -> EventDataCategory
-        rv = "default"  # type: Any
-        event = self.get_event()
-        if event is not None:
-            rv = get_event_data_category(event)
+        ty = self.headers.get("type")
+        if ty in ("session", "attachment", "transaction"):
+            return ty
+        elif ty == "event":
+            return "error"
         else:
-            ty = self.headers.get("type")
-            if ty in ("session", "attachment"):
-                rv = ty
-        return rv
+            return "default"
 
     def get_bytes(self):
         # type: (...) -> bytes
         return self.payload.get_bytes()
 
     def get_event(self):
         # type: (...) -> Optional[Event]
-        if self.headers.get("type") == "event" and self.payload.json is not None:
+        """
+        Returns a regular event if there is one. Transaction events are exempt
```

Comment:
What do you mean by "transaction events are exempt from this"?

Reply:
Nevermind. I reverted the changes to this method as they were not necessary.